common/src/main/java/com/microsoft/hydralab/performance/parsers/AndroidMemoryInfoResultParser.java

```diff
@@ -40,18 +41,56 @@ public PerformanceTestResult parse(PerformanceTestResult performanceTestResult)
             return null;
         }
         List<PerformanceInspectionResult> inspectionResults = performanceTestResult.performanceInspectionResults;
+        double[] averageMemoryInfo = new double[MEM_INFO_LENGTH];
+        Arrays.fill(averageMemoryInfo, 0.0);
+        int validDataSize = 0;
         for (PerformanceInspectionResult inspectionResult : inspectionResults) {
             File logFile = inspectionResult.rawResultFile;
             long[] memInfos = parseRawResultFile(logFile);
             inspectionResult.parsedData = buildMemoryInfo(inspectionResult.inspection.appId, inspectionResult.inspection.description, inspectionResult.timestamp, memInfos);
+            if (isValidMem(memInfos)) {
+                validDataSize++;
+                updateAverageMem(averageMemoryInfo, memInfos, validDataSize);
+            }
         }
 
+        performanceTestResult.setResultSummary(buildAverageMemoryInfo(averageMemoryInfo, inspectionResults.get(0)));
+
         return performanceTestResult;
     }
 
+    private void updateAverageMem(double[] averageMemoryInfo, long[] memInfoArray, int index) {
+        if (memInfoArray == null) {
+            return;
+        }
+        for (int i = 0; i < MEM_INFO_LENGTH; i++) {
+            averageMemoryInfo[i] = averageMemoryInfo[i] * index / (index + 1) + (double) memInfoArray[i] / (index + 1);
+        }
+    }
+
+    private boolean isValidMem(long[] memInfo) {
+        if (memInfo == null) {
+            return false;
+        }
+
+        for (long i : memInfo) {
+            if (i != -1) {
+                return true;
+            }
+        }
+        return false;
```

Comment:
```suggestion
        for (long i : memInfo) {
            if (i == -1) {
                return false;
            }
        }
        return true;
```

Reply:
Or -1 should be excluded in the calculation of the average.

Reply:
There may be a chance that some metrics are missing on some devices, but the total PSS is valid. So the criteria for judging invalid data can be looser.